src/mimefactory.rs

```diff
@@ -954,14 +954,28 @@ impl<'a, 'b> MimeFactory<'a, 'b> {
         );
 
         // Message is sent as text/plain, with charset = utf-8
-        let main_part = PartBuilder::new()
+        let mut main_part = PartBuilder::new()
             .header((
                 "Content-Type".to_string(),
                 "text/plain; charset=utf-8; format=flowed; delsp=no".to_string(),
             ))
             .body(message_text);
         let mut parts = Vec::new();
 
+        // add HTML-part, this is needed only if a HTML-message from a non-delta-client is forwarded;
+        // for simplificity and to avoid conversion errors, we're generating the HTML-part from the original message.
+        if self.msg.has_html() {
+            if let Some(org_msg_id) = self.msg.param.get_int(Param::Forwarded) {
+                let org_msg_id = MsgId::new(org_msg_id as u32);
```

Comment:
afaik this is quietly a panic if someone gets bad stuff in the db, perhaps:
```suggestion
                let org_msg_id = MsgId::new(org_msg_id.try_into()?);
```
as this function can already return an error?

Reply:
+1